src/twister2/twister_config.py

```diff
@@ -121,6 +127,9 @@ def get_platform(self, name: str) -> PlatformSpecification:
                 return platform
         raise KeyError(f'There is not platform with identifier: {name}')
 
+    def verify_platforms_existence(self, platform_names: list[str]):
```

Comment:
Do you consider changing this method into `staticmethod`? Thanks to this you can avoid creating `_verify_platforms_existence` as a separate function and then use it in `_get_selected_platforms` as `TwisterConfig.verify_platforms_existence(selected_platforms, platforms)`

```python
@staticmethod
def verify_platforms_existence(
    platform_names_to_verify: list[str],
    platforms: list[PlatformSpecification]
) -> None:
    """Verify if platform names are correct, if not - raise exception"""
    ...
```

Reply:
But `_get_selected_platforms ` method is called only once from `@classmethod - create`. I added  `_verify_platforms_existence ` as a separate method, because in `create` method, we do not have `TwisterConfig` instance yet, so we cannot use `self.platforms` to look for.

Reply:
I thought about something like that:
```python
@dataclass
class TwisterConfig:
    """Store twister configuration to have easy access in test."""

    ...

    @staticmethod
    def verify_platforms_existence(
        platform_names_to_verify: list[str],
        platforms: list[PlatformSpecification]
    ) -> None:
        """Verify if platform names are correct, if not - raise exception"""
        platform_names = [p.identifier for p in platforms]
        for platform in platform_names_to_verify:
            if platform not in platform_names:
                msg = f'Unrecognized platform - {platform}.'
                logger.error(msg)
                raise TwisterConfigurationException(msg)

def _get_selected_platforms(config: pytest.Config) -> list[str]:
    """Return list of selected platforms"""

    ...  

    if platform_filter:
        selected_platforms = list(set(platform_filter))
        TwisterConfig.verify_platforms_existence(selected_platforms, platforms)
    
    ...
```

And it works, because static method don't need to be called on class object. It can be called as 
`TwisterConfig.verify_platforms_existence(selected_platforms, platforms)` 
in example above.

On the other hand in when you use `verify_platforms_existence` in `specification_processor.py` file, you will need to pass `platforms` as a separate argument each time, for example:
`self.twister_config.verify_platforms_existence(tc_integration_platforms, self.twister_config.platforms)`
and this is quite "ugly". 

So, all I think, that it can stay as you proposed in this PR - there is no one best solution.

Reply:
Please check now. What do you think about moving platform verifying to __post_init__ method?